File: app/src/androidTest/java/com/nytsample/android/APITests.java
package com.nytsample.android;

import com.nytsample.android.data.api.NYTimesAPI;
import com.nytsample.android.data.model.NewsFeedResponseModel;

import org.junit.Test;

import static org.junit.Assert.assertTrue;

public class APITests {
    @Test
    public void test_getNewsFeedSuccess() {
        NYTimesAPI.getNewsFeed(0, new NYTimesAPI.NewsFeedCallback() {
            @Override
            public void onSuccess(NewsFeedResponseModel newsFeedResponseModel) {
                assertTrue(newsFeedResponseModel != null);
                assertTrue(newsFeedResponseModel.numOfResults > 0);
            }

            @Override
            public void onFailure(String error) {

            }
        });
    }

    @Test
    public void test_getNewsFeedFail() {
        NYTimesAPI.getNewsFeed(5, new NYTimesAPI.NewsFeedCallback() {
            @Override
            public void onSuccess(NewsFeedResponseModel newsFeedResponseModel) {

            }

            @Override
            public void onFailure(String error) {
                assertTrue(!error.isEmpty());
            }
        });
    }
}

File: app/src/androidTest/java/com/nytsample/android/APITests.kt
package com.nytsample.android

import com.nytsample.android.data.api.NYTimesApi
import com.nytsample.android.data.model.NewsFeedResponseModel
import org.junit.Assert
import org.junit.Test

class APITests {
    @Test
    fun test_getNewsFeedSuccess() {
        NYTimesApi.getNewsFeed(callback = object : NYTimesApi.Companion.NewsFeedCallback {
            override fun onSuccess(newsFeedResponseModel: NewsFeedResponseModel) {
                Assert.assertTrue(newsFeedResponseModel != null)
                Assert.assertTrue(newsFeedResponseModel.numOfResults > 0)
            }

            override fun onFailure(error: String) {
            }
        })
    }

    @Test
    fun test_getNewsFeedFail() {
        NYTimesApi.getNewsFeed(5, object : NYTimesApi.Companion.NewsFeedCallback {
            override fun onSuccess(newsFeedResponseModel: NewsFeedResponseModel) {
            }

            override fun onFailure(error: String) {
                Assert.assertTrue(error.isNotEmpty())
            }
        })
    }
}